### ISSUE TITLE
Custom tracker - multiple frames at once problem

### ISSUE THREAD

**Issue by michalzachwieja** [traccar/traccar#2271]
Hello
I'm building my own GSM-GPS tracker hardware. 
Right now I'm using XEXUN or GPS103 protocols to send data to traccar and it works OK.
But I need to send few frames (multiple devices) per one connection  (cos reconnecting over AT commands takes time- +300ms)  something like this (IMEI is different for each line) :

```
160804204341,,GPRMC,204341.000,A,5000.0000,N,01800.0000,E,0.13,,040816,,,D*6D,F,,imei:1234,00,0.00,F:2.12V,0,123,00000,000,00,0000,0000 
160804204341,,GPRMC,204341.000,A,5000.0000,N,01800.0000,E,0.13,,040816,,,D*6D,F,,imei:5678,00,0.00,F:2.12V,0,123,00000,000,00,0000,0000 
160804204341,,GPRMC,204341.000,A,5000.0000,N,01800.0000,E,0.13,,040816,,,D*6D,F,,imei:8765,00,0.00,F:2.12V,0,123,00000,000,00,0000,0000 
160804204341,,GPRMC,204341.000,A,5000.0000,N,01800.0000,E,0.13,,040816,,,D*6D,F,,imei:4321,00,0.00,F:2.12V,0,123,00000,000,00,0000,0000 
```

and there is a problem with that cos traccar update only one device and then closes the tcp link (and re connecting to tcp server takes another +1s) 
I was trying different combinations (new lines after each frame, no new lines etc) but nothing works over GPRS
BUT if I connect through some tcp client like realterm and chose "send file" option It works OK! (no delays between lines etc. just text file containing lines as above ) 

Any ideas ? Maybe there is other protocol that can parse multiple frames at once ?
Thanks
Michael

**Comment by tananaev** [comment]
Traccar does not close TCP connection.

**Comment by michalzachwieja** [comment]
So what does? GSM modem and realterm ? I don't think so. There must be some thing wrong when parsing the frame. 
Please try it : get realterm and connect to port 5006 then send 2 frames in one time : 
160804204341,,GPRMC,204341.000,A,5000.0000,N,01800.0000,E,0.13,,040816,,,D6D,F,,imei:8765,00,0.00,F:2.12V,0,123,00000,000,00,0000,0000 160804204341,,GPRMC,204341.000,A,5000.0000,N,01800.0000,E,0.13,,040816,,,D6D,F,,imei:4321,00,0.00,F:2.12V,0,123,00000,000,00,0000,0000

**Comment by tananaev** [comment]
OK, I fixed the problem with frame decoder (see f8d0544edb04745afdd481eaf6424d8601784472), but messages themselves are invalid. There is no asterisk before checksum.

**Comment by michalzachwieja** [comment]
thanks! I'll test that
